packages/nextjs/src/config/webpack.ts

```diff
@@ -336,23 +336,40 @@ export function constructWebpackConfigFunction(
-          // TODO(v9): Remove this warning and print warning in case source map deletion is auto configured
-          if (!isServer && !userSentryOptions.sourcemaps?.deleteSourcemapsAfterUpload) {
-            // eslint-disable-next-line no-console
-            console.warn(
-              "[@sentry/nextjs] The Sentry SDK has enabled source map generation for your Next.js app. If you don't want to serve Source Maps to your users, either set the `sourcemaps.deleteSourcemapsAfterUpload` option to true, or manually delete the source maps after the build. In future Sentry SDK versions `sourcemaps.deleteSourcemapsAfterUpload` will default to `true`. If you do not want to generate and upload sourcemaps, set the `sourcemaps.disable` option in `withSentryConfig()`.",
+          // enable source map deletion if not explicitly disabled
+          if (!isServer && userSentryOptions.sourcemaps?.deleteSourcemapsAfterUpload === undefined) {
+            logger.warn(
+              '[@sentry/nextjs] Source maps will be automatically deleted after being uploaded to Sentry. If you want to keep the source maps, set the `sourcemaps.deleteSourcemapsAfterUpload` option to false in `withSentryConfig()`. If you do not want to generate and upload sourcemaps at all, set the `sourcemaps.disable` option to true.',
             );
+            userSentryOptions.sourcemaps = {
+              ...userSentryOptions.sourcemaps,
+              deleteSourcemapsAfterUpload: true,
+            };
           }
 
-          // `hidden-source-map` produces the same sourcemaps as `source-map`, but doesn't include the `sourceMappingURL`
-          // comment at the bottom. For folks who aren't publicly hosting their sourcemaps, this is helpful because then
-          // the browser won't look for them and throw errors into the console when it can't find them. Because this is a
-          // front-end-only problem, and because `sentry-cli` handles sourcemaps more reliably with the comment than
-          // without, the option to use `hidden-source-map` only applies to the client-side build.
-          if (isServer || userNextConfig.productionBrowserSourceMaps) {
-            newConfig.devtool = 'source-map';
-          } else {
-            newConfig.devtool = 'hidden-source-map';
+          // Source maps can be configured in 3 ways:
+          // 1. (next config): productionBrowserSourceMaps
+          // 2. (next config): experimental.serverSourceMaps
+          // 3. custom webpack configuration
+          //
+          // We only update this if no explicit value is set
+          // (Next.js defaults to `false`: https://github.com/vercel/next.js/blob/5f4f96c133bd6b10954812cc2fef6af085b82aa5/packages/next/src/build/webpack/config/blocks/base.ts#L61)
+          if (!newConfig.devtool) {
+            logger.info(
+              `[@sentry/nextjs] Automatically enabling source map generation for ${
+                isServer ? 'server' : 'client'
```

Comment:
```suggestion
                runtime
```

this may get called 3 times (server, client, edge)